webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -132,56 +255,94 @@ trait PromiseGenerationService extends Loggable {
 
     import HooksImplicits._
 
+    val maxParallelism = {
+      // We want to limit the number of parallel execution and threads to the number of core/2 (minimum 1) by default.
+      // This is taken from the system environment variable because we really want to be able to change it at runtime.
+      def threadForProc(mult: Double): Int = {
+        Math.max(1, (Runtime.getRuntime.availableProcessors * mult).ceil.toInt)
+      }
+      val t = try {
+        getMaxParallelism().getOrElse("x0.5") match {
```

Comment:
Value of maxParallelism should be directly a double or a float (and maybe should be name parallelismFactor) with value between 0 and 1 (maybe not limited by 1 but i do fell it would cause problem to have more than 1)

Reply:
Ok we can use it to set a fix value ...